File: wakakusa/demo/mixer.rb
# State an app keeps and a field that writes into it.
require "wakakusa"

class Mixer
  def initialize
    @volume = 5
    @title = "untitled"
    @muted = false
  end

  def view
    cells = [
      text("#{@title} — vol #{@volume}", size: 16.0),
      row(
        button("+1") { @volume += 1 },
        button("mute") { @muted = true },
        button("unmute") { @muted = false },
        spacing: 8.0
      ),
    ]
    cells.push(text("(muted)", size: 12.0, color: "#8a8f98")) if @muted
    cells.push(text_field(@title, placeholder: "title") { |t| @title = t })
    column(*cells, spacing: 10.0, padding: 14.0)
  end
end

run(Mixer.new, title: "mixer")

File: gomamochi/demo/mixer.go
// State an app keeps and a field that writes into it.
package main

import (
	"fmt"

	. "github.com/i2y/yokan/gomamochi"
)

type Mixer struct {
	volume int
	title  string
	muted  bool
}

func (m *Mixer) View() Element {
	cells := []Element{
		Text(fmt.Sprintf("%s — vol %d", m.title, m.volume)).Size(16),
		Row(
			Button("+1").OnClick(func() { m.volume += 1 }),
			Button("mute").OnClick(func() { m.muted = true }),
			Button("unmute").OnClick(func() { m.muted = false }),
		).Spacing(8),
	}
	if m.muted {
		cells = append(cells, Text("(muted)").Size(12).Color("#8a8f98"))
	}
	cells = append(cells, TextField(m.title).Placeholder("title").OnChange(func(t string) { m.title = t }))
	return Column(cells...).Spacing(10).Padding(14)
}

func main() {
	Run(&Mixer{volume: 5, title: "untitled"}, Title("mixer"))
}
